plumbing/mpool/pool.go

```diff
@@ -0,0 +1,49 @@
+package mpool
+
+import (
+	"context"
+
+	"gx/ipfs/QmR8BauakNcBa3RbE4nbQu76PDiJgoQgz8AJdhJuiU4TAw/go-cid"
+	"gx/ipfs/QmVRxA4J3UPQpw74dLrQ6NJkfysCA1H4GU28gVpXQt9zMU/go-libp2p-pubsub"
+
+	"github.com/filecoin-project/go-filecoin/core"
+	"github.com/filecoin-project/go-filecoin/plumbing/msg"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+// Pool interfaces to the core message pool
+type Pool struct {
+	pool   *core.MessagePool
+	pubsub *pubsub.PubSub
+}
+
+// New builds a new Pool.
+func New(pool *core.MessagePool, pubsub *pubsub.PubSub) *Pool {
+	return &Pool{pool: pool, pubsub: pubsub}
+}
+
+// Pending lists un-mined messages in the pool
+func (ls *Pool) Pending(ctx context.Context, messageCount uint) ([]*types.SignedMessage, error) {
+	pending := ls.pool.Pending()
+	if len(pending) < int(messageCount) {
+		subscription, err := ls.pubsub.Subscribe(msg.Topic)
```

Comment:
maybe subscribe before calling pending in case a message arrives in between these calls as written?

Reply:
I've added an additional call to pending before the first `Next()`, preserving the fast path that avoids subscribing at all unless necessary.